### PR TITLE
image: Add image datasource

### PR DESCRIPTION
**Pull request by maveonair**
This pull requests adds the image datasource as discussed in https://github.com/lxc/terraform-provider-incus/issues/96.

**Example**

```hcl
data "incus_image" "my_image" {
  name = "my_image" # is the alias of the image, fingerprint would also be possible to set instead
}

output "fingerprint" {
  value = data.incus_image.my_image.fingerprint
}

resource "incus_instance" "instance1" {
  image = data.incus_image.my_image.fingerprint
  name  = "instance1"
}
```

**What's been done**
- Added image datasource
- Added docs for image datasource

### CODE REVIEW

File: docs/data-sources/image.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+* `name` - *Optional* - Name of the image.
+
+* `fingerprint` - *Optional* - Fingerprint of the image.
+
+* `type` - *Optional* - Type of image to cache. Must be one of `container` or `virtual-machine`. Defaults to `container`.
+
+* `architecture` - *Optional* - The image architecture (e.g. x86_64, aarch64). See [Architectures](https://linuxcontainers.org/incus/docs/main/architectures/) for all possible values.
+
+* `project` - *Optional* - Name of the project where the profile will be stored.
```
Comment (stgraber) [inline docs/data-sources/image.md]: Should that be "image" rather than "profile" here?

File: docs/data-sources/image.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+* `remote` - *Optional* - The remote in which the resource will be created. If
+  not provided, the provider's default remote will be used.
```
Comment (stgraber) [inline docs/data-sources/image.md]: Nothing is really getting created here, should that be tweaked?

File: internal/image/datasource_image.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/lxc/incus/v6/shared/api"
+	"github.com/lxc/terraform-provider-incus/internal/errors"
+	"strings"
+
+	provider_config "github.com/lxc/terraform-provider-incus/internal/provider-config"
```
Comment (stgraber) [inline internal/image/datasource_image.go:15]: Import grouping is wrong :)

Should be:
```
    "context"
	"fmt"
	"strings"

	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
	"github.com/hashicorp/terraform-plugin-framework/datasource"
	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
	"github.com/hashicorp/terraform-plugin-framework/types"
	"github.com/lxc/incus/v6/shared/api"

	"github.com/lxc/terraform-provider-incus/internal/errors"
	provider_config "github.com/lxc/terraform-provider-incus/internal/provider-config"
```

File: docs/data-sources/image.md (hunk 1 above)
Comment (stgraber) [reply docs/data-sources/image.md]: I see this gets fixed in a follow-up commit, can you squash the two together?

File: docs/data-sources/image.md (hunk 2 above)
Comment (stgraber) [reply docs/data-sources/image.md]: I see this gets fixed in a follow-up commit, can you squash the two together?

File: internal/image/datasource_image.go (hunk 3 above)
Comment (maveonair) [reply internal/image/datasource_image.go:15]: Thanks! I somehow always miss that :(

**Review by stgraber — APPROVED** [review]

**Pull request merged**